bundles/org.eclipse.ui.navigator/src/org/eclipse/ui/navigator/WizardActionGroup.java

```diff
@@ -177,6 +182,29 @@ public WizardActionGroup(IWorkbenchWindow aWindow,
 
 	}
 
+	/**
+	 * Set a filter for {@link IWizardDescriptor}s
+	 *
+	 * @param descriptorFilter the filter to set, might be <code>null</code> if no
+	 *                         filtering is desired.
+	 *
+	 * @since 3.11
+	 */
+	public void setFilter(Predicate<IWizardDescriptor> descriptorFilter) {
```

Comment:
What about having it final and set in a new constructor?

Reply:
I first started with a constructor parameter, but this then ended up in many constructs with long arguments list so I go with a setter as it seems more descriptive.

Reply:
I still believe 1 new constructor is actually better than 1 setter as immutability is usually making maintenance and debug much easier. A long argument list may not be convenient, but it's still less error prone than mutability.

Reply:
done